Automattic/wp-calypso
# Add `getThemeType` selector

Related to https://github.com/Automattic/dotcom-forge/issues/2438

## Proposed Changes

In order to reuse the same component for rendering the theme type badges in the design picker and in the theme showcase (https://github.com/Automattic/dotcom-forge/issues/2438) we need a single way to infer the theme type.

This PR introduces a new `getThemeType` selector to centralize that logic rather than duplicating it across the codebase. The new selector will be used by a follow-up PR which will refactor how the theme type badges are rendered.

For now, the `/with-theme` flow has been refactored to use the new selector.

## Testing Instructions

- Open the Calypso live link below in a incognito/private browser session.
- Go to `/themes.
- Select a free, a premium, a WooCommerce, a dotOrg and a Marketplace theme and copy the URL of the "Pick design" button.
- To simplify things, switch to a browser session where you're logged in and paste the URL.
- Pick the needed plan for every theme and create the site.
- Make sure you land in the relevant page to activate the themes you selected above.
